[WIP] Remove the catch-up function from the test code

MichaelKim20 opened the pull request:
I checked late after seeing intermittent errors in some test codes when github's CI was executed due to the catch-up function.

I was able to remove the catch-up function from the test code to solve this problem.

Geod24 commented:
If it fails in the CI, it's probably because there's a bug. Avoiding the call will not solve it, just hide it.

MichaelKim20 committed:
Change how catch-up runs

Instead of using `this.pending` when invoking catch-up function, call function 'catchup' directly.
Instead of nested Promise, Promise-Chain was used.
This ensures that the test code can be executed normally after all pre-work is completed.

MichaelKim20 committed:
Change to separate fake Agora and Stoa when preparing for test

MichaelKim20 commented:
This is a problem caused by Mocha waiting only for the first Promise in the process of processing Promise chain.
So I plan to make an issue of this.

MichaelKim20 commented:
```TypeScript
describe ('Test of Promise in Mocha', () =>
{
    before("begin", () => {
        function p1():Promise<void>
        {
            return new Promise<void>((resolve, reject) => {
                setTimeout(() => {
                    console.log('Promise 1');
                    resolve();
                }, 100);
            })
        }

        function p2():Promise<void>
        {
            return new Promise<void>((resolve, reject) => {
                setTimeout(() => {
                    console.log('Promise 2');
                    resolve();
                }, 200);
            });
        }

        function p3(): Promise<void>
        {
            return new Promise<void>((resolve, reject) => {
                setTimeout(() => {
                    console.log('Promise 3');
                    resolve();
                }, 300);
            });
        }

        return p1()
            .then(() => { return p2; })
            .then(() => { return p3; });
    });

    it ('it 2', () =>
    {
        console.log("it 2");
    })
});

describe ('Test of Promise in Mocha2', () =>
{
    it ('it 1', () =>
    {
        function p1():Promise<void>
        {
            return new Promise<void>((resolve, reject) => {
                setTimeout(() => {
                    console.log('Promise 1');
                    resolve();
                }, 100);
            })
        }
        function p2():Promise<void>
        {
            return new Promise<void>((resolve, reject) => {
                setTimeout(() => {
                    console.log('Promise 2');
                    resolve();
                }, 200);
            });
        }

        function p3(): Promise<void>
        {
            return new Promise<void>((resolve, reject) => {
                setTimeout(() => {
                    console.log('Promise 3');
                    resolve();
                }, 300);
            });
        }


        return p1()
            .then(() => { return p2; })
            .then(() => { return p3; });
    });

    it ('it 2', () =>
    {
        console.log("it 2");
    })
});
```
Result is 
Promise 1
it 2

MichaelKim20 commented:
Since this is new, I made it proceed in issue #190
